src/mode/modeHandler.ts

```diff
@@ -1095,8 +1096,25 @@ export class ModeHandler implements vscode.Disposable {
         case ">":
           if (letterToTheLeft === "<") { return true; }
           break;
-        default:
-          return false;
+      }
+
+      // This section is mostly for vscode auto closing brackets without a user inserting them
+      const letterToTheRight = TextEditor.getLineAt(vimState.cursorPosition).text[vimState.cursorPosition.character];
+      if (letterToTheRight !== undefined) {
+        switch (key) {
+          case "{":
```

Comment:
I think this idea would also let you DRY up this code a little bit. It's a little repetitive right now.